.github/workflows/build-phar.yml

```diff
@@ -0,0 +1,63 @@
+# Invoking this pipeline requires additional permissions, so must be invoked
+# in a way to pass those permissions on, e.g.:
+#
+#  build-phar:
+#    permissions:
+#      contents: read
+#      id-token: write
+#      attestations: write
+#    uses: ./.github/workflows/build-phar.yml
+
+name: "Build the PIE PHAR"
+
+on:
+  workflow_call:
+
+permissions:
+  contents: read
+
+jobs:
+  build-phar:
+    runs-on: ${{ matrix.operating-system }}
+    strategy:
+      matrix:
+        operating-system:
+          - ubuntu-latest
+        php-versions:
+          - '8.1'
+    permissions:
+      # id-token:write is required for build provenance attestation.
+      id-token: write
+      # attestations:write is required for build provenance attestation.
+      attestations: write
+    steps:
+      - name: Setup PHP
+        uses: shivammathur/setup-php@v2
+        with:
+          coverage: none
+          tools: composer, box
+          php-version: "${{ matrix.php-version }}"
+      - uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+      # Ensure some kind of previous tag exists, otherwise box fails
+      - run: git describe --tags HEAD || git tag 0.0.0
+      - uses: ramsey/composer-install@v3
+      - name: Build PHAR
+        run: box compile
+      - name: Check the PHAR executes
+        run: php pie.phar --version
+      # It does not make sense to do this for PR builds, not do contributors
```

Comment:
```suggestion
      # It does not make sense to do this for PR builds, nor do contributors
```

Reply:
And perhaps move the comment right above the `if` for clarity.

Reply:
Thanks :+1: